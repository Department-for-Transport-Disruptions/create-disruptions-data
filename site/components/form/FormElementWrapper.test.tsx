import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import FormElementWrapper from "./FormElementWrapper";

describe("FormElementWrapper", () => {
    it("should render correctly with no errors", () => {
        const { asFragment } = render(
            <FormElementWrapper errors={[]} errorId={""} errorClass={""}>
                <div />
            </FormElementWrapper>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render correctly around an input with errors", () => {
        const { asFragment } = render(
            <FormElementWrapper
                errors={[
                    {
                        errorMessage: "There was an error",
                        id: "test-div",
                    },
                ]}
                errorId={"test-div"}
                errorClass={"govuk-input--error"}
            >
                <input id="test-div" />
            </FormElementWrapper>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
