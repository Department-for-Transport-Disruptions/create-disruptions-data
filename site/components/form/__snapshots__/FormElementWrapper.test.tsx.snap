// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormElementWrapper > should render correctly around an input with errors 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <span
      class="govuk-error-message"
      id="test-div-error"
    >
      <span
        class="govuk-visually-hidden"
      >
        Error: 
      </span>
      There was an error
    </span>
    <input
      aria-describedby="test-div-error"
      class="govuk-input--error"
      id="test-div"
    />
  </div>
</DocumentFragment>
`;

exports[`FormElementWrapper > should render correctly with no errors 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div />
  </div>
</DocumentFragment>
`;
